### ISSUE TITLE
Feature: Interleaved Practice Module

### ISSUE THREAD

**Issue by tankh99** [tankh99/learnie#4]
Interleaves notes from different, but related areas together

**Proposed implementation**
- Add hierarchical, user-modifiable tags for notes. Tags are mandatory if you want to include them in the Practice module
- User clicks on a "Practice" button and he will be prompted to choose which tag he wants to practice. 
   - Note: in future, we will add Practice support for questions with no tags. Reminder to create a new issue once this is completed
- User is shown a set of flashcard questions where these flashcard questions are grouped under a single parent category.

Depends on #3
